File: A2Z/js/sorting/1/bubble-sort.mjs
class Solution {
  // Function to sort the array using bubble sort algorithm.
  bubbleSort(arr) {
    for (let i = 0; i < arr.length; i++) {
      for (let j = 1; j < arr.length - i; j++) {
        const prevJ = j - 1;

        if (arr[prevJ] > arr[j]) {
          const tmp = arr[j];
          arr[j] = arr[prevJ];
          arr[prevJ] = tmp;
        }
      }
    }

    return arr;
  }
}

export default Solution;

File: A2Z/go/sorting/1/bubble_sort.go
package sortone

func BubbleSort(input []int) []int {
	for i := 0; i < len(input); i++ {
		for j := 1; j < len(input)-i; j++ {
			prevJ := j - 1
			if input[prevJ] > input[j] {
				tmp := input[j]
				input[j] = input[prevJ]
				input[prevJ] = tmp
			}
		}
	}

	return input
}
